feat: Map additional typescript script targets and module kinds (#6694)
SAP/cloud-sdk-js: .changeset/metal-rivers-fry.md

---

> "@sap-cloud-sdk/generator-common": patch

Author: ```suggestion
"@sap-cloud-sdk/generator-common": minor
```

---

[impr] Extended `parseScriptTarget` to include ES2023, ES2024, and ES2025 targets.

> Extended `parseModuleKind` to include UMD, System, ES2022, Node18, Node20, and Preserve module kinds.

Author: ```suggestion
[impr] Extended script target mapper to handle ES2023, ES2024, and ES2025 targets.
Extended module kind mapper to handle UMD, System, ES2022, Node18, Node20, and Preserve module kinds.
```

> Extended module resolution mapping to handle Bundler.

Reviewer: ```suggestion
[impr] Extend script target mapper to handle ES2023, ES2024, and ES2025 targets.
Extend module kind mapper to handle UMD, System, ES2022, Node18, Node20, and Preserve module kinds.
Extend module resolution mapping to handle Bundler.
```
